compiler/ztests/quoted-type.yaml

```diff
@@ -0,0 +1,19 @@
+script: |
+  zq  -z 'is(<"@foo">)' in.zson
+  echo ===
+  zq  -z 'type "@foo" = {x:int64} type "Y Z" = {y:"@foo"} is(<"Y Z">)' in.zson
```

Comment:
It's a little unfortunate for readability that a double-quoted string in a Zed program is no longer necessarily a Zed string value, but I don't have any good ideas about how to maintain that status quo. Guess it might just be the price of progress!